packages/bpk-component-image/src/BpkImage.js

```diff
@@ -42,39 +42,62 @@ type BpkImageProps = {
 
 type ImageProps = {
   altText: string,
-  hidden: boolean,
+  hidden: ?boolean,
   onImageLoad: ?() => mixed,
 };
 
-const Image = (props: ImageProps): Node => {
-  const { hidden, altText, onImageLoad, ...rest } = props;
+class Image extends Component<ImageProps> {
+  img: ?HTMLImageElement;
 
-  const imgClassNames = [getClassName('bpk-image__img')];
+  static defaultProps = {
+    hidden: false,
+  };
 
-  if (hidden) {
-    imgClassNames.push(getClassName('bpk-image__img--hidden'));
+  constructor() {
+    super();
+    this.img = null;
   }
 
-  return (
-    <img
-      className={imgClassNames.join(' ')}
-      alt={altText}
-      onLoad={onImageLoad}
-      {...rest}
-    />
-  );
-};
+  componentDidMount() {
+    if (this.img && this.img.src && this.img.complete) {
+      if (this.props.onImageLoad) {
+        this.props.onImageLoad();
+      }
+    }
+  }
+
+  setImgRef(el) {
+    this.img = el;
+  }
+
+  render() {
+    const { hidden, altText, onImageLoad, ...rest } = this.props;
+
+    const imgClassNames = [getClassName('bpk-image__img')];
+
+    if (hidden) {
+      imgClassNames.push(getClassName('bpk-image__img--hidden'));
+    }
+
+    return (
+      <img
+        className={imgClassNames.join(' ')}
+        alt={altText}
+        onLoad={onImageLoad}
+        ref={el => this.setImgRef(el)}
```

Comment:
One thing I wanted to check, do do we need to do `this.setImageRef = this.setImageRef.bind(this)` for this to work? I had that in my original code I tested in our project but Flow was complaining here so I took it out.

Reply:
No, but you could change it to `ref={this.setImgRef}` in which case it would need to be bound. We don't explicitly bind instead preferring 

```javascript
setImageRef = (el) => {
  this.img = el;
}
```

Which achieves the same thing

Reply:
Just out of curiosity, why you prefer this approach?

In the team we took at look at pros/cons between arrow function and explicitly bind and didn't get much 😅 
Performance wise seems to be more or less the same, arrow function is less code but on the other side not all functions in the class can be created with arrow functions, leading to an inconsistent file... so we're still undecided 😂

Reply:
Arrow functions don't go into the prototype, which means that you create them every time the component is instantiated. That could have a performance penalty, but in general is negligible.
They can also be problematic for testing but this depends on your testing approach. In general use or not arrow functions here is a matter of taste. If you use the object method approach and you need to use it as a callback (and only in this case) you need to bind the this.

Reply:
We think it's nicer than having a bunch of `binds` in the constructor. I'm not sure about the performance between creating arrow functions in render, using instance properties with arrow functions or calling `bind` in the constructor. In any case it's probably a matter of premature optimization to worry too much about it